js/src/tags.js

```diff
@@ -0,0 +1,467 @@
+        var reset_button = template.find('.' + this.button_class_reset);
+        var delete_button = template.find('.' + this.button_class_delete);
+
+        template.find('.' + this.class_editable_field).prop('disabled', !edit_mode);
+
+        edit_button.text(edit_mode ? this.localised_save_edit : this.localised_edit);
+        delete_button.text(this.localised_delete);
+        reset_button.text(this.localised_reset);
+
+        edit_button.hide();
+        reset_button.hide();
+        delete_button.hide();
+
+        if ((!new_item && (deleted || modified)) || (new_item && edit_mode)) {
+            reset_button.show();
+        }
+        if (!deleted) {
+            edit_button.show();
+        }
+        if ((new_item && !edit_mode) || (!deleted && !edit_mode && !modified)) {
+            delete_button.show();
+        }
+
+        template.attr('data-id', id);
+
+        this.fillRowFields(id, template, item);
+
+        return template;
+    };
+
+    ListableItemsAdmin.prototype.fillRowFields = function (id, template, item) {
+        var self = this;
+        template.find('.' + this.class_editable_field).each(function () {
+            var field = $(this);
+            var type = field.attr('type');
+            var name = field.attr('data-name');
+            //TODO: Add lodash 'has' for deeply complex objects, once lodash exists.
+            var value = (name in item ? item[name] : null);
+
+            if (type === 'text') {
+                field.val(value);
+            } else if (type === 'checkbox') {
+                field.attr('checked', value);
+            } else {
+                console.log('Unsupported field type ' + type);
+            }
+
+            if (id in self.item_errors && name in self.item_errors[id]) {
+                var error_banner = field.next('.' + self.class_editable_field_error);
+                if (error_banner.length > 0) {
+                    error_banner.text(self.item_errors[id][name]);
+                }
+            }
+        });
+    };
+
+    ListableItemsAdmin.prototype.checkModified = function () {
+        if (Object.keys(this.modified_items).length !== 0 || this.deleted_items_ids.length > 0) {
+            this.button_save.show();
+            this.button_cancel.show();
+        } else {
+            this.button_save.hide();
+            this.button_cancel.hide();
+        }
+    };
+
+    ListableItemsAdmin.prototype.checkPageNavigation = function () {
+        var page_number = this.page;
+        if (page_number === 1) {
+            this.button_previous.hide();
+        } else {
+            this.button_previous.show();
+        }
+
+        if (page_number < this.page_count) {
+            this.button_next.show();
+        } else {
+            this.button_next.hide();
+        }
+    };
+
+    ListableItemsAdmin.prototype.getFieldsFromTemplate = function () {
+        var template = $($.parseHTML(this.html_template.html()));
```

Comment:
Another template that could benefit from pre-compilation and placeholder usage.

Reply:
Fixed in ef1176accf5c411a39af7103186807ffe773720a